src/utils/routes/Routes.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { FC, ComponentType } from 'react';
+import { RouteProps } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { RootStore } from '../../redux/reducers/rootReducer';
+import PrivateRoute from './PrivateRoute';
+
+export interface LoggedInRouteProps extends RouteProps {
+  component: ComponentType<any>;
+}
+
+export const LoggedInRoute: FC<LoggedInRouteProps> = ({ component, ...rest }) => {
+  const user = useSelector((store: RootStore) => store.user);
```

Comment:
får meg til å tenke at vi kunne hentet `user` bare i `AppRoutes` og sendt det ned som prop for å slippe å slå opp i store på nytt for hver route. Men vi har jo gjort det på samme måte med `useFormikContext` også, så dette er uansett mest konsistent. finner ikke noe om best practice på dette heller :/

Reply:
tenkte på det jeg også, men tror ikke det er bedre å sende ned som props uansett, for det vil jo bli like mange renders?

Reply:
blir sikkert like mange renders ja, men det hadde blitt 1 lookup i redux store i stedet for 8 (per render). men tror jeg også liker det bedre slik det står nå anywayz :P